source/Shuffle.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Akshay Agarwal <agarwal.akshay.akshay8@gmail.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+#include "Shuffle.h"
+
+
+void 
+Swap(int* a, int* b)
+{
+	int temp = *a;
+	*a = *b;
+	*b = temp;
+}
+
+
+void 
+Randomize(int* arr, int n)
+{
+	srand ( time(NULL) );
+
+	for(int i = n - 1; i > 0; i--) {
+		int j = rand() % (i+1);
```

Comment:
gcc2 required variables to be declared prior to code (c89).  I suspect this code will fail if you were to try and build it with gcc2.  Looking at the haikuports recipe, the 1.0.1 version was marked as compiling on both gcc2 and gcc4.